From 843fdc234745ddcc8c61ed63440f989f28b8039d Mon Sep 17 00:00:00 2001
From: Fabian Vogt <fvogt@suse.com>
Date: Fri, 26 Aug 2016 09:18:44 +0200
Subject: [PATCH] dt: assign uart0 to BT and uart1 to pin headers

Patch-mainline: v4.14-rc1
Git-commit: 4188ea2aeb6dd8f99ab77662f463e41bc464a704 (partial)
References: bsc#983145

This maps uart0 to the on-board BT module and uart1 to the header.
Both are tested and work fine.

Signed-off-by: Fabian Vogt <fvogt@suse.com>
---
 arch/arm64/boot/dts/broadcom/bcm2837-rpi-3-b.dts | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/broadcom/bcm2837-rpi-3-b.dts b/arch/arm64/boot/dts/broadcom/bcm2837-rpi-3-b.dts
index c09707b..3b22035 100644
--- a/arch/arm64/boot/dts/broadcom/bcm2837-rpi-3-b.dts
+++ b/arch/arm64/boot/dts/broadcom/bcm2837-rpi-3-b.dts
@@ -25,9 +25,16 @@
 	};
 };
 
+/* uart0 communicates with the BT module */
+&uart0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart0_ctsrts_gpio32 &gpclk2_gpio43>;
+};
+
+/* uart1 is mapped to the pin header */
 &uart1 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&uart1_gpio32>;
+	pinctrl-0 = <&uart1_gpio14>;
 	status = "okay";
 };
 
-- 
2.6.2

